compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/SharedVariablesLowering.kt

```diff
@@ -106,7 +106,6 @@ class SharedVariablesLowering(val context: BackendContext) : FunctionLoweringPas
                     if (declaration !in sharedVariables) return declaration
 
                     val newDeclaration = context.sharedVariablesManager.declareSharedVariable(declaration)
-                    newDeclaration.parent = irFunction
```

Comment:
This could affect JS and Native

Reply:
I am not sure about native, but the JS sharedVariablesManager sets this field correctly. The problem with this line is that it doesn't set the correct parent.

I'll check what kotlin-native does.

Reply:
Nice catch! This does indeed have to be fixed in kotlin-native. I've created a separate PR for this: https://github.com/JetBrains/kotlin-native/pull/2855

Reply:
@max-kammerer I changed it to keep the assignment here. At worst it's redundant but at least it won't break anything.